fedimint-core/src/db/mod.rs

```diff
@@ -467,13 +587,13 @@ impl<'isolated, 'parent, T: Send + Encodable + 'isolated> IDatabaseTransaction<'
 
 #[doc = " A handle to a type-erased database implementation"]
 pub struct DatabaseTransaction<'a> {
-    tx: Box<dyn IDatabaseTransaction<'a>>,
+    tx: Box<dyn ISingleUseDatabaseTransaction<'a>>,
```

Comment:
can we make this `&'a mut ISingleUseDatabaseTransaction<'a>` and save some boxing?

Reply:
Ideally this would be `tx: &dyn IDatabaseTransaction<'a>,`. Here we have no need to know about the wrapper anymore, no?

Reply:
IDatabaseTransaction still has `commit(self)` right?

Reply:
So, I think the issue is that `DatabaseTransaction` needs to hold on to the actual boxxed `ISingleDatabaseTransaction`, otherwise (when holding onto a reference) we are creating the `DatabaseTransaction` and returning a value referencing a temporary object.

I think this can be worked around if we introduce a new public wrapper type instead of reusing `DatabaseTransaction` for everything. 

Example with bad name: `NewModuleDatabaseTransaction` could be the db transaction we pass to the modules, it would not have a `commit_tx` function and it would hold a `&mut dyn ISingleUseDatabaseTransaction` that referenced the existing db transaction. `DatabaseTransaction` would stay the way it is right now and always hold a `Box`.

Perhaps I can do this in a different PR if you guys agree.